src/test/java/com/exasol/dbbuilder/dialects/mysql/MySQLDatabaseObjectCreationIT.java

```diff
@@ -0,0 +1,158 @@
+package com.exasol.dbbuilder.dialects.mysql;
+
+import static com.exasol.dbbuilder.dialects.mysql.MySqlGlobalPrivilege.CREATE_ROLE;
+import static com.exasol.dbbuilder.dialects.mysql.MySqlGlobalPrivilege.CREATE_USER;
+import static com.exasol.dbbuilder.dialects.mysql.MySqlObjectPrivilege.DELETE;
+import static com.exasol.dbbuilder.dialects.mysql.MySqlObjectPrivilege.SELECT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.sql.*;
+
+import org.junit.jupiter.api.*;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.exasol.dbbuilder.dialects.*;
+
+@Tag("integration")
+@Testcontainers
+class MySQLDatabaseObjectCreationIT {
+    private static final String MYSQL_DOCKER_IMAGE_REFERENCE = "mysql:8.0.20";
+    @Container
+    private static final MySQLContainer<?> container = new MySQLContainer<>(MYSQL_DOCKER_IMAGE_REFERENCE)
+            .withUsername("root").withPassword("");
+    private MySqlObjectFactory factory;
+    private Connection adminConnection;
+
+    @BeforeEach
+    void beforeEach() throws SQLException {
+        this.adminConnection = container.createConnection("");
+        this.factory = new MySqlObjectFactory(container.createConnection(""));
+    }
+
+    @Test
+    void testCreateSchema() {
+        final Schema schema = this.factory.createSchema("THE_SCHEMA");
+        assertSchemaExistsInDatabase(schema.getName());
+    }
+
+    private void assertSchemaExistsInDatabase(final String schemaName) {
+        final String sql = "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = \"" + schemaName
+                + "\"";
+        assertObjectExistsInDatabase(sql, schemaName);
+    }
+
+    private void assertObjectExistsInDatabase(final String sql, final String objectName) {
+        try (final PreparedStatement objectExistenceStatement = this.adminConnection.prepareStatement(sql)) {
+            final ResultSet resultSet = objectExistenceStatement.executeQuery();
+            assertAll( //
+                    () -> assertThat(resultSet.next(), equalTo(true)), //
+                    () -> assertThat(resultSet.getString(1), equalTo(objectName)));
+        } catch (final SQLException exception) {
+            throw new AssertionError("Unable to determine existence of object: " + objectName, exception);
+        }
+    }
+
+    @Test
+    void testCreateTable() {
```

Comment:
We need to remove code duplication between the tests. You could use the `Schema` interface and do the exact same test for both dialects.
Create an abstract test class with all common tests, derive concrete test classes with the rest from that class.

Reply:
done